javascript/collapsible.js

```diff
@@ -18,3 +18,11 @@ function collapseAllGroups() {
       tableRows[i].style.display = 'none';
   }
 }
+function collapsefile(CurrObj) {
```

Comment:
In general, just follow camel case like the other code uses : `collapsefile` -> `collapseFile`, `CurrObj` -> `currObj`

BUT: I don't like `onclick` and calling this function in the HTML. Why did you introduce it at all? Your solution with registering a click-handler in JS afterwards looks far better. Why is that commented, didn't work? I prefer those, not `onclick`.

Reply:
They were two though process for that. You had requested not to have it inline. I though this what you meant. That is my mistake. The other was i wanted to use the pre-existing collapsing icons that were there in the templates. I wanted to reuse them, Thought that was the reason you asked to not make it inline. 
I do not know how to work with the CSS or add icons. I thought of reusing what is already there.

Reply:
Have a look at this 
![image](https://user-images.githubusercontent.com/4973182/89547605-98f6fd80-d823-11ea-9502-426cdfd02c24.png)
This has a class name open and closed. So when it is collapsed it refers to "th.closed" and when it is expanded it refers to "th.open". This is in CSS file for the template. This is common for all templates. Unfortunately all templates do not have toggle icons like calm.

Reply:
With the new commit I have removed the collapsefile and kept the original solution :). Will think of applying the icon in some different way.

Reply:
If icons are available already, things are pretty easy: Change the HTML-template to ALWAYS add the "open" class. This is [done in some templates already](https://github.com/websvnphp/websvn/blob/master/templates/calm/index.tmpl#L30):

      [websvn-test:opentree]
              class="open"
      [websvn-else]
              class="closed"
      [websvn-endtest]

Afterwards you change usage of `toggle` in your JS to [toggleClass](https://api.jquery.com/toggleclass/#toggleClass-function-state). This way you can check if the currently applied class is `open` or `close` and change accordingly. Should be something like the following:

    $(this) .closest("table").find("tbody")
            .toggleClass(function(idx, oldClass, state)
    {
      return oldClass == "open" ? "closed" : "open";
    });

You need to get syntax and interaction of `oldClass` vs. `state` correct on your own.

Reply:
I have successfully done for the calm template but for the other two there are issues.
I implemented in a sligthly different way the code is below:
```
$("thead").find("th").on("click", function() {
    $(this).get(0).className = ($(this).get(0).className == 'open') ? 'closed' : 'open';
    $(this).closest("table").find("tbody").toggle();
    
});
```
I defaulted the class to open in the html template.
Then i added the following in the CSS
```
thead th.open {
	background-image:url(images/toggledown.png);
	background-repeat:no-repeat;
}
thead th.closed {
	background-image:url(images/toggleup.png);
	background-repeat:no-repeat;
}
```
This is how it looks in calm template
![image](https://user-images.githubusercontent.com/4973182/89677987-01b9a500-d90c-11ea-8b46-9b544e75dde7.png)

This all is in the commit.
This same i could not apply for the other two templates.
For elegant the comparison tables uses a repeated image to generate the background
![image](https://user-images.githubusercontent.com/4973182/89678087-33cb0700-d90c-11ea-92af-71d056161b71.png)
For bluegrey there is no template specific icon present. If you can provide some solution for those two after looking in the commit i can do that change as well.

Reply:
Please rename the toggle-icons to `toggle-down|up.png`, as that is better readable.

If you are not interested in the other templates yourself, you might simply ignore them for now. In theory, different templates simply can look differently, so the approach with icons for `calm` might not apply to other templates.

If you care about those templates, have a look at the download-icon of `Elegant`:

https://github.com/websvnphp/websvn/blob/master/templates/Elegant/images/download.png

That might be used for both templates lacking toggle-icons, as the color scheme somewhat fits both. Simply copy and rename the file to follow the naming scheme `toggle-down|up.png` and rotate the image to get the missing toggle-icon.

As for the background of `Elegant`, I would try to create a new background image simply containing the toggle-icons and apply that wherever you want to toggle: `bg-table-header-toggle-down|up.png`, `bg-table-divider-toggle-down|up.png`

Reply:
I have done the changes for the calm template and provided the commit for it. 
There seems to be a problem with the BlueGrey Template. I have checked in the master copy in git(Not mine). Whenever i do compare in bluegrey it shows an error as below. This error is not there in any other template. 
The error snapshot
![image](https://user-images.githubusercontent.com/4973182/89706673-8bf21f80-d985-11ea-9829-a4aa43cf4249.png)
The working snapshot
![image](https://user-images.githubusercontent.com/4973182/89706693-bd6aeb00-d985-11ea-8941-e9d11ad2552a.png)

Shall i raise an issue for this?

Reply:
I have raised an issue for this.